### ISSUE TITLE
"No usable pokeballs found" when there are 200 balls

### ISSUE THREAD

**Issue by alextj** [PokemonGoF/PokemonGo-Bot#3215]
I have a bag full of Ultraballs, so I configured bot not to keep any Pokeballs or Greatballs.
```
[PokemonGoBot] [INFO] PokeBalls: 0 | GreatBalls: 0 | UltraBalls: 200
```

### Expected Behavior
Use Ultraball to catch pokemon.

### Actual Behavior
Bot doesn't use Ultraballs, instead just gives up:
```
[PokemonCatchWorker] [INFO] [pokemon_appeared] A wild Tauros appeared! [CP 75] [Potential 0.53] [A/D/S 8/1/15]
[PokemonCatchWorker] [INFO] [no_pokeballs] No usable pokeballs found!
```

### Steps to Reproduce
- 1. Setup config to keep only Ultraballs and better.
```
        "type": "RecycleItems",
        "config": {
          "item_filter": {
            "Pokeball":       { "keep" : 0 },
            "Greatball":      { "keep" : 0 },
```
- 2. Run bot and let it try to catch Pokemons.

### Other Information
OS: OS X 10.11.5
Git Commit: 4b16e9da985e689d57beae8bf18878f4d872bdc4
Python Version: Python 2.7.12
Config file is valid JSON

The reason I don't want to keep Pokeballs is because I have tons of Ultraballs and I notice many pokemons are not caught and vanish when using Pokeballs. The bot should use Ultraballs.

**Comment by mrthrowaway** [comment]
as a workaround, I guess you could add an 'any' rule under the 'vip' section that matches all pokemon

**Comment by nburn42** [comment]
It is saving Ultraballs for vip pokemon only.

**Comment by k4n30** [comment]
#3271 #3294

**Comment by k4n30** [comment]
Will be solved when #3421 is merged

**Comment by k4n30** [comment]
Merged
